### ISSUE TITLE
System.OverflowException: Array dimensions exceeded supported range.

### ISSUE THREAD

**Issue by Razkan** [Starcounter/Home#381]
Hi,

This just happened when I tried to start my test database this morning. Since it is a test database I have nothing important on it so I'll just remove and create a new one, but before I remove it I will store a local copy of it if you want to investigating it.

Version=2.3.1.7786

```
System.OverflowException: Array dimensions exceeded supported range.
   at Starcounter.Db.CreateTable(TableDef tableDef, TableDef inheritedTableDef) in C:\TeamCity\BuildAgent\work\sc-10919\Level1\src\Starcounter\Db.cs:line 176
   at Starcounter.Binding.TableCreate.Eval() in C:\TeamCity\BuildAgent\work\sc-10919\Level1\src\Starcounter\Binding\TableCreate.cs:line 47
   at Starcounter.Hosting.Package.CreateOrUpdateDatabaseTable(TypeDef typeDef) in C:\TeamCity\BuildAgent\work\sc-10919\Level1\src\Hosting\Starcounter.Hosting\Package.cs:line 453
   at Starcounter.Hosting.Package.UpdateDatabaseSchemaAndRegisterTypes(String fullAppId, TypeDef[] unregisteredTypeDefs, TypeDef[] allTypeDefs) in C:\TeamCity\BuildAgent\work\sc-10919\Level1\src\Hosting\Starcounter.Hosting\Package.cs:line 355
   at Starcounter.Hosting.Package.ProcessWithinCurrentApplication(Application application, ApplicationDirectory applicationDir) in C:\TeamCity\BuildAgent\work\sc-10919\Level1\src\Hosting\Starcounter.Hosting\Package.cs:line 266
   at Starcounter.Hosting.Package.Process() in C:\TeamCity\BuildAgent\work\sc-10919\Level1\src\Hosting\Starcounter.Hosting\Package.cs:line 139
   at Starcounter.Hosting.Package.Process(IntPtr hPackage) in C:\TeamCity\BuildAgent\work\sc-10919\Level1\src\Hosting\Starcounter.Hosting\Package.cs:line 56
   at StarcounterInternal.Hosting.Processor.RunMessageLoop(Void* hsched) in C:\TeamCity\BuildAgent\work\sc-10919\Level1\src\Hosting\Starcounter.Hosting\Processor.cs:line 89
HResult=-2146233066
```

**Comment by k-rus** [comment]
Thank you for reporting. 
Seems that you reach the limit of number of supported columns in a table.

Can you describe what actually you were doing? Have you made any changes to the schema?

cc @chrhol @miyconst

**Comment by miyconst** [comment]
>This just happened when I tried to start my test database this morning. Since it is a test database I have nothing important on it so I'll just remove and create a new one, but before I remove it I will store a local copy of it if you want to investigating it.

@Razkan please provide more information, such as:

- Did the database work before?
- What did you do which lead to this crash?
- Does the crash go away if you drop and re-create the database?
- Does it happen when you start an app (which) or when starting the database without any apps?

**Comment by Razkan** [comment]
@k-rus 

- Started the database through visual studio, nothing unusuallyfrom how I start other databases. I have done changes to the schema, what those changes are can be hard to track since i'd estimate it is about 4-7 days I last time I started the database (or sooner). Mostly it is changes to the properties (added properties, removed properties & changed enum structure used by database entities), don't recall I have done anything other.

@miyconst 

1. See above answer
2. See above answer
3. If you mean remove and create a new database (with the same name and settings) by drop and re-create the database, then yes
4. I removed the database, bit late to test this one unfortunetly

**Comment by chrhol** [comment]
I think this is the same problems as we detected earlier, that dropped columns is not really removed and is just ignored during upgrade which leads to incorrect behaviour later. So same problem as #368 but a strange exception instead of AV. 

I'm about to add this check in the tableupgrade functionality, to get an exception earlier with a proper message that removed database-properties has to be dropped as well.

**Comment by chrhol** [comment]
Gathered from the other issues I have an idea on how to reproduce it.

**Comment by miyconst** [comment]
@chrhol please note, it has to be fixed in `2.3.2`.

**Comment by chrhol** [comment]
The added verification of removed properties, that are not dropped is in place. Now a proper error should be thrown if a mismatch if databaseschema contains columns that is not available in the clr-class. 

Since @Razkan already got past this problem, I will close this issue.
